### PR TITLE
Endo integration

### LINKED ISSUE #7090: Need endo release, update agoric-sdk to depend on that release
We have a large overhang of code in agoric-sdk which is redundant with code in recent endo. Once updated, we could remove the redundancy and the hazards of divergence.

See
 - https://github.com/Agoric/agoric-sdk/pull/6792
 - https://github.com/Agoric/agoric-sdk/pull/7081
 - https://github.com/endojs/endo/pull/1502
   - would fix https://github.com/Agoric/agoric-sdk/issues/6570
https://github.com/Agoric/agoric-sdk/pull/6816
 - #7129 

Speedups we may need that are also blocked on this:
 - https://github.com/endojs/endo/pull/1503
 - https://github.com/endojs/endo/pull/1504
 - https://github.com/endojs/endo/pull/1528 (+ https://github.com/Agoric/agoric-sdk/pull/7263 )

Debugging support we need, that's blocked on this:
https://github.com/endojs/endo/pull/1513 (though see #7270 )

cc @ivanlei

### PR DESCRIPTION
**Pull request by michaelfig**
closes: #7090

## Description

- Sync Endo versions from NPM, removing the assimilated `@endo` patch-packages files.
- Reflect introduction of `fromCapData` and `toCapData` in `@endo/marshal`
- More Typescript fixes for improvements to `@endo` typings
- Update golden hashes for new xsnap supervisor snapshots

### Security Considerations

None known.

### Scaling Considerations

None known.

### Documentation Considerations

Should be mainly compatible.

### Testing Considerations

None necessary.

### CODE REVIEW

**Comment by warner** [comment]
FWIW, message-patterns.js is probably overenthusiastically parallelized, if AVA doesn't have any sort of limits, it might try to launch 110 kernels at the same time, each with six-ish vats. The one saving grace is that `defaultReapInterval: 'never'`, so they won't be doing BOYD and engineGC() after every delivery.

The last one (`forward to fake zoe`) is lacking a `defaultManagerType: 'xs-worker'`, unlike the other two tests in that same time. I see that I left a commented-out switch in `vats.target` to control the `managerType`, and the comments make me think I was investigating an XS misbehavior. I *think* I left everything using `local` workers for speed, but given the GC intentions of the file, I think it ought to have a `defaultManagerType: 'xs-worker'`, and that the error you observed might be our old villian/pal #5575 because we were using `local` worker.

**Comment by Chris-Hibbert** [comment]
I don't think the change in the automaticRefund test represents a bug.

The contract exits the seat, which informs ZCF.  ZCF asks Zoe to exit the seat, but doesn't await the result. When the test has the seat call `tryExit()`,  Zoe still doesn't know that the seat has exited, but it waits for Zcf's response while trying to exit. ZCF does know that the seat has exited, so it does complain.

So either "seat has already exited" or "seat has been exited" are acceptable errors.

File: packages/vats/src/core/client-behaviors.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ async function createLocalBundle(vats, devices, vatAdminSvc, vatPowers) {
     http: httpRegCallback,
     vattp: makeVattpFrom(vats),
   });
-  /** @type { typeof bundleP } */
+  /** @type {Promise<typeof bundleP>} */
```
Comment (turadg) [inline packages/vats/src/core/client-behaviors.js]: using `deeplyFulfilledObject` might work better here and not require a type annotation

Comment (michaelfig) [reply packages/vats/src/core/client-behaviors.js]: Thanks!  That works well.

File: yarn.lock [hunk 2]
```diff
@@ -12190,6 +12235,11 @@ ses@^0.18.0:
   resolved "https://registry.yarnpkg.com/ses/-/ses-0.18.0.tgz#9bc1f17712e2f2cf9a38c83156197a6fe722822e"
   integrity sha512-WMEOltdIqYk11uZiD+0ANnTLcpAQXyLAPm5NneJSbSPEzxhSn4G7B/bs3dtSOcEicAlFI6WFXqXKDMu9BBCj+w==
 
+ses@^0.18.3:
```
Comment (turadg) [inline yarn.lock:11962]: this adds a ses but doesn't take one away. `npx yarn-deduplicate` would clean it up. When I ran it 297 lines were removed. Many of them were dupes of Endo deps so I think this is worth doing.

**Review by turadg — APPROVED** [review]
LGTM as long as the fixups are squashed

**Pull request merged**